### PR TITLE
✋feat: send transaction by fee rate

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: lib/ckb/transaction_size.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+module CKB
+  module TransactionSize
+    SERIALIZED_OFFSET_BYTESIZE = 4
+
+    class << self
+      def base_size
+        68 + SERIALIZED_OFFSET_BYTESIZE
+      end
+
+      def every_cell_deps
```
Comment (shaojunda) [inline lib/ckb/transaction_size.rb]: `cell_dep_size` may be better?

Comment (classicalliu) [reply lib/ckb/transaction_size.rb]: It's every `cell_dep` in tx size, cell_dep size + extra 4 bytes

Comment (shaojunda) [reply lib/ckb/transaction_size.rb]: Why `cell_dep_size` and `header_dep_size` are fixed value?

Comment (classicalliu) [reply lib/ckb/transaction_size.rb]: `struct`, all data fields are fixed.

Comment (shaojunda) [reply lib/ckb/transaction_size.rb]: it’s a bit strange to add a plural form after `every` and `cell_deps` and `header_deps` are field in `transaction`.

Comment (classicalliu) [reply lib/ckb/transaction_size.rb]: updated

File: lib/ckb/wallet.rb [hunk 2]
```diff
@@ -70,9 +71,19 @@ def get_balance
     # @param data [String ] "0x..."
     # @param key [CKB::Key | String] Key or private key hex string
     # @param fee [Integer] transaction fee, in shannon
-    def generate_tx(target_address, capacity, data = "0x", key: nil, fee: 0, use_dep_group: true)
+    def generate_tx(target_address, capacity, data = "0x", key: nil, fee_rate: 0, use_dep_group: true)
```
Comment (shaojunda) [inline lib/ckb/wallet.rb]: I am not sure if using fee_rate as default is a good choice. Let's take a look at other people's opinions.

Comment (ashchan) [inline lib/ckb/wallet.rb]: Param doc wasn't updated. (`fee`)

Comment (classicalliu) [reply lib/ckb/wallet.rb]: updated

**Comment by classicalliu** [comment]
Now updated to support both fee and fee_rate, please review again @shaojunda @ashchan

File: lib/ckb/wallet.rb [hunk 3]
```diff
@@ -137,20 +151,28 @@ def generate_tx(target_address, capacity, data = "0x", key: nil, fee_rate: 0, us
...
     # @param fee_rate [Integer] shannons per KB
     #
     # @return [CKB::Type::OutPoint]
-    def deposit_to_dao(capacity, key: nil, fee_rate: 0)
+    def deposit_to_dao(capacity, key: nil, fee: 0, fee_rate: 0)
       key = get_key(key)
 
+      mode = if fee == 0 && fee_rate > 0
+               :fee_rate
+             else
```
Comment (shaojunda) [inline lib/ckb/wallet.rb]: may be you can extract a separate method

Comment (classicalliu) [reply lib/ckb/wallet.rb]: updated

**Pull request closed without merging**